src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,204 @@
+        '[class.ux-badge-medium]': 'badgeSize === "medium"',
+        '[class.ux-badge-large]': 'badgeSize === "large"',
+        '[class.ux-badge-above]': 'badgeVerticalPosition === "above"',
+        '[class.ux-badge-below]': 'badgeVerticalPosition === "below"',
+        '[class.ux-badge-after]': 'badgeHorizontalPosition === "after"',
+        '[class.ux-badge-before]': 'badgeHorizontalPosition === "before"',
+    },
+})
+export class BadgeDirective implements OnChanges, OnDestroy /*, AfterContentChecked*/ {
+    private readonly _className = 'ux-badge';
+    private readonly _darkColor: ThemeColor = ThemeColor.parse('#000');
+    private readonly _lightColor: ThemeColor = ThemeColor.parse('#FFF');
+    private _badgeElement: HTMLElement | undefined;
+    private _isNumber: boolean;
+    private _truncatedText: boolean = false;
+
+    /**
+     * Directive parameter that sets the content of the badge
+     */
+    private _badgeDisplayContent: string;
+
+    @Input('uxBadge')
+    get badgeContent(): string {
+        return this._badgeContent;
+    }
+    set badgeContent(s: string) {
+        const subject = s.toString().trim();
+        this._isNumber = /^\d+$/.test(subject);
+        this._badgeContent = subject;
+    }
+    private _badgeContent: string;
+
+    /**
+     * Define the badge background color
+     */
+    private readonly _defaultBackgroundColor: ThemeColor = this._darkColor;
+
+    @Input()
+    get badgeColor(): string {
+        return this._badgeColor.toHex();
+    }
+    set badgeColor(color: string) {
+        this._badgeColor = ThemeColor.parse(this._colorService.resolve(color));
+    }
+    private _badgeColor: ThemeColor = this._defaultBackgroundColor;
+
+    /**
+     * Aria description value for accessibility devices
+     */
+    @Input() badgeAriaDescription: string;
+
+    /**
+     * Set the badge vertical position in relation to the parent element
+     */
+    private _defaultVerticalPosition: BadgeVerticalPosition = 'above';
+
+    @Input()
+    private badgeVerticalPosition: BadgeVerticalPosition = this._defaultVerticalPosition;
+
+    /**
+     * Set the badge horizontal position in relation to the parent element
+     */
+    private _defaultHorizontalPosition: BadgeHorizontalPosition = 'after';
+
+    @Input()
+    private badgeHorizontalPosition: BadgeHorizontalPosition = this._defaultHorizontalPosition;
+
+    /**
+     * Set if the badge overlaps parent content or flows after parent
+     */
+    @HostBinding('class.ux-badge-overlap')
+    @Input()
+    badgeOverlap: boolean = false;
+
+    /**
+     * Max value that can be displayed - if string measures no. of characters,
+     * if number checks against exact number not no. of characters
+     */
+    @Input() badgeMaxValue: number;
+
+    /**
+     * Badge size (based on CSS styles)
+     */
+    private _defaultBadgeSize: BadgeSize = 'medium';
+
+    @Input()
+    private badgeSize: BadgeSize = this._defaultBadgeSize;
```

Comment:
`_defaultBadgeSize` is only ever used here so we can just change this to:


```
@Input() badgeSize: BadgeSize = 'medium';
```

I believe `_defaultVerticalPosition` and `_defaultHorizontalPosition` and perhaps a few others may also be the same